sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/integration/tests/HIFIOElasticIT.java

```diff
@@ -0,0 +1,158 @@
+  private static HIFTestOptions options;
+  private static final int SIZE = 10;
+
+  @BeforeClass
+  public static void setUp() {
+    PipelineOptionsFactory.register(HIFTestOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(HIFTestOptions.class);
+    LOGGER.info("Pipeline created successfully with the options");
+  }
+
+  /**
+   * This test reads data from the Elasticsearch instance and verifies whether data is read
+   * successfully.
+   */
+  @Test
+  public void testHifIOWithElastic() {
+    Pipeline pipeline = TestPipeline.create(options);
+    Configuration conf = getConfiguration(options);
+    PCollection<KV<Text, LinkedMapWritable>> esData =
+        pipeline.apply(HadoopInputFormatIO.<Text, LinkedMapWritable>read().withConfiguration(conf));
+    PCollection<Long> count = esData.apply(Count.<KV<Text, LinkedMapWritable>>globally());
+    PAssert.thatSingleton(count).isEqualTo((long) SIZE);
+
+    PCollection<LinkedMapWritable> values = esData.apply(Values.<LinkedMapWritable>create());
+    MapElements<LinkedMapWritable, String> transformFunc =
+        MapElements.<LinkedMapWritable, String>via(new SimpleFunction<LinkedMapWritable, String>() {
+          @Override
+          public String apply(LinkedMapWritable mapw) {
+            Text text = (Text) mapw.get(new Text("scientist"));
+            return text != null ? text.toString() : "";
+          }
+        });
+
+    PCollection<String> textValues = values.apply(transformFunc);
+    List<String> expectedResults =
+        Arrays.asList("Faraday", "Newton", "Galilei", "Maxwell", "Pasteur", "Copernicus", "Curie",
+            "Bohr", "Darwin", "Einstein");
+    PAssert.that(textValues).containsInAnyOrder(expectedResults);
+    pipeline.run().waitUntilFinish();
+  }
+
+  /**
+   * This test reads data from the Elasticsearch instance based on a query and verifies if data is
+   * read successfully.
+   */
+  @Test
+  public void testHifIOWithElasticQuery() {
+    Pipeline pipeline = TestPipeline.create(options);
+    Configuration conf = getConfiguration(options);
+    String query =
+        "{" + "  \"query\": {"
+            + "  \"match\" : {"
+            + "    \"scientist\" : {"
+            + "      \"query\" : \"Newton\","
+            + "      \"type\" : \"boolean\""
+            + "    }"
+            + "  }"
+            + "  }"
+            + "}";
+    conf.set(ConfigurationOptions.ES_QUERY, query);
+    PCollection<KV<Text, LinkedMapWritable>> esData =
+        pipeline.apply(HadoopInputFormatIO.<Text, LinkedMapWritable>read().withConfiguration(conf));
+    PCollection<Long> count = esData.apply(Count.<KV<Text, LinkedMapWritable>>globally());
+    PAssert.thatSingleton(count).isEqualTo((long) 1);
+    pipeline.run().waitUntilFinish();
+  }
+
+  /**
+   * Set the Elasticsearch configuration parameters in the Hadoop configuration object.
+   * Configuration object should have InputFormat class, key class and value class to be set
```

Comment:
"All Configuration objects used by HadoopInputFormat must set InputFormat class, key class and value class"